packages/ingest/s3.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const aws = require('@cumulus/common/aws');
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const urljoin = require('url-join');
+
+module.exports.s3Mixin = superclass => class extends superclass {
+
+  /**
+   * Downloads a given url and upload to a given S3 location
+   * @return {Promise}
+   * @private
+   */
+  async sync(_path, bucket, key, filename) {
+    const source = _path.replace(/^s3:\//, '');
+    const params = {
+      Bucket: bucket,
+      CopySource: _path.replace(/^s3:\//, ''),
+      Key: path.join(key, filename),
+      ACL: 'private'
+    };
+    await aws.s3().copyObject(params).promise();
+    return urljoin('s3://', bucket, key, filename);
+  }
+
+  /**
+   * Upload a file to S3
+   *
+   * @param {string} bucket - the S3 bucket to upload to
+   * @param {string} key - the base path of the S3 key
+   * @param {string} filename - the filename to be uploaded to
+   * @param {string} tempFile - the location of the file to be uploaded
+   * @returns {Promise<string>} - the S3 URL that the file was uploaded to
+   */
+  async upload(bucket, key, filename, tempFile) {
+    const fullKey = `${key}/${filename}`;
+
+    await aws.s3().putObject({
+      Bucket: bucket,
+      Key: fullKey,
+      Body: fs.createReadStream(tempFile)
+    }).promise();
+
+    return `s3://${bucket}/${fullKey}`;
+  }
+
+  /**
+   * Downloads the file to disk, difference with sync is that
+   * this method involves no uploading to S3
+   * @return {Promise}
+   * @private
+   */
+  async download(_path, filename) {
```

Comment:
The first argument is used, so it should not start with an underscore. An underscore indicates an argument that must be present for positional reasons, but is not used.